### PR TITLE
Preserve embeddings across rebuilds and fix store race condition

### PR DESCRIPTION
**Pull request by mrosnerr**
## Summary

- **Preserve embeddings during rebuild**: `rebuildFromFiles()` now saves embedding data before `DELETE FROM notes` and restores it after re-indexing, preventing loss of computed vectors during any maintenance action that involves a rebuild (rebuild, migrate-layout, full)
- **Auto-backfill after rebuild**: The `rebuild` maintenance action now automatically backfills embeddings for any notes missing them, using the shared `backfillEmbeddings()` helper
- **Fix handleStore race condition**: `handleStore()` is now async and `await`s embedding generation instead of using a fire-and-forget `void` promise that could be lost if the process exited before completion

## Root Cause

`rebuildFromFiles()` performed `DELETE FROM notes` then re-inserted rows without carrying over `embedding` / `embedding_model` columns (which only exist in SQLite, not in markdown files). Any rebuild wiped all embeddings — explaining why only 1/211 notes had embeddings despite local MiniLM being enabled.

## Changes

| File | Change |
|------|--------|
| `src/storage/NoteRepository.ts` | Save embeddings before DELETE, restore after INSERT loop |
| `src/tool-handlers.ts` | Extract `backfillEmbeddings()` helper; add auto-embed to rebuild; make `handleStore` async |
| `src/mcp-server.ts` | `await` the now-async `handleStore()` |
| `tests/edge-cases.test.ts` | Update tests for async `handleStore` |
| `tests/mcp-tools.test.ts` | Update tests for async `handleStore` |

## Test Results

385 pass, 0 fail, 39 skip (424 total)

### CODE REVIEW

File: src/storage/NoteRepository.ts [hunk 1]
```diff
@@ -1092,6 +1097,17 @@ export class NoteRepository {
       }
     }
 
+    if (savedEmbeddings.length > 0) {
+      const restoreStmt = this.db.prepare(
+        'UPDATE notes SET embedding = ?, embedding_model = ? WHERE id = ?'
+      );
+      for (const saved of savedEmbeddings) {
+        if (uniqueIds.has(saved.id)) {
+          restoreStmt.run(saved.embedding, saved.embedding_model, saved.id);
```
[the remark this answers is not in the document]
Comment (mrosnerr) [reply src/storage/NoteRepository.ts:1812]: Fixed in commits 8af157e and aabb64a. The restore loop now computes a SHA-256 hash of (title + summary + content) before the DELETE and compares it against the freshly-indexed content after rebuild. Embeddings are only restored when the hash matches — changed notes are left with embedding IS NULL so the backfill picks them up. A regression test in tests/integration.test.ts covers this path.

File: src/tool-handlers.ts [hunk 2]
```diff
@@ -1229,14 +1258,26 @@ export async function handleMaintain(args: MaintainArgs, repo: NoteRepository, c
     case 'rebuild': {
       if (gitVersioning) await gitVersioning.checkpoint('Pre-rebuild snapshot');
       const result = repo.rebuildFromFiles();
+      let output = `Indexed ${result.indexed} notes, ${result.errors} errors\nRebuild complete.`;
+      if (embeddingConfig) {
+        try {
+          const embResult = await backfillEmbeddings(repo, embeddingConfig);
+          if (embResult.requested > 0) {
```
[the remark this answers is not in the document]
Comment (mrosnerr) [reply src/tool-handlers.ts:1272]: Valid catch. `repo.store()` uses `INSERT OR REPLACE INTO notes` without `embedding`/`embedding_model` columns, so `rebuildProjectIndex()`/`appendProjectLog()` do wipe embeddings on any index/log notes that were just backfilled.

Practically the impact is limited to structural notes (index/log) which are auto-generated navigation surfaces and poor candidates for semantic search anyway — but it is wasted embedding work and the fix is a trivial reorder: move the backfill block after the navigation rebuild loop.

Will fix in a follow-up commit.

[the remark this answers is not in the document]
Comment (mrosnerr) [reply src/tool-handlers.ts:1272]: Fixed in `382751f` — moved the backfill block after `rebuildProjectIndex`/`appendProjectLog`/`updateGlobalNavigation` so structural note embeddings aren't immediately wiped by navigation updates.
